### PR TITLE
Add .asf.yaml file

### LINKED ISSUE #544: [NEW] Add .asf.yaml file
So that we can configure repo settings under apache org.

### PR DESCRIPTION
**Pull request by tisonkun**
This closes #544.

### CODE REVIEW

**Comment by tisonkun** [comment]
cc @git-hulk @Hexiaoqiao

**Review by git-hulk — DISMISSED** [review]
+1, LGTM

**Comment by ShooterIT** [comment]
On other branch, could we use `rebase` merge button?
I used `rebase` merge  when I release new version, such as #468

**Comment by tisonkun** [comment]
@ShooterIT merge button config applied on the whole repository. There is no branch level configure. But I'm OK to enable rebase and merge option as long as we keep linear history at unstable and release branch.

File: .asf.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+github:
+  description: >-
+    Kvrocks is a distributed key value NoSQL database that uses RocksDB as storage engine
+    and is compatible with Redis protocol.
+  homepage: https://kvrocks.apache.org/
+  enabled_merge_buttons:
+    squash:  true
+    merge:   false
+    rebase:  false
```
Comment (tisonkun) [inline .asf.yaml]:
```suggestion
    rebase:  true
```

**Review by ShooterIT — APPROVED** [review]

**Review by git-hulk — APPROVED** [review]

File: .asf.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# specific language governing permissions and limitations
+# under the License.
+
+# For more information, see https://cwiki.apache.org/confluence/display/INFRA/Git+-+.asf.yaml+features.
+
+github:
+  description: >-
+    Kvrocks is a distributed key value NoSQL database that uses RocksDB as storage engine
+    and is compatible with Redis protocol.
+  homepage: https://kvrocks.apache.org/
```
Comment (Hexiaoqiao) [inline .asf.yaml:24]: I think we should use https://kvrocks.incubator.apache.org rather than https://kvrocks.apache.org/ when incubating.

**Review by Hexiaoqiao — COMMENTED** [review]
Leave one nit comment inline. Others look good to me.

File: .asf.yaml (hunk 2 above)
Comment (tisonkun) [reply .asf.yaml:24]: I see almost all incubator projects doesn't have the `incubator` section. For example,

* https://inlong.apache.org/
* https://eventmesh.apache.org/

Comment (Hexiaoqiao) [reply .asf.yaml:24]: Sure. According this guide(https://incubator.apache.org/guides/sites.html), both of them are accepted, and therefore some podlings point to p.incubator.a.o and others point to p.a.o.
LGTM. +1. Thanks.

**Review by Hexiaoqiao — APPROVED** [review]

**Pull request merged**